doc/develop/test/twister_statuses.rst

```diff
@@ -0,0 +1,206 @@
+
+.. list-table:: Twister Statuses
+   :widths: 10 10 70 10
+   :header-rows: 1
+
+   * - In-code
+     - In-text
+     - Description
+     - Where applicable
+   * - NOTRUN
+     - not run
+     - The test wasn't run because it was not runnable in current configuration.
+       It was, however, built correctly.
+     - Suite, Case
+   * - BLOCK
+     - blocked
+     - The test was not run because of an error or crash in the test suite.
+     - Case
+   * - FILTER
+     - filtered
+     - A static or runtime filter has eliminated the test from the list of tests to use.
+     - Suite, Case
+   * - ERROR
+     - error
+     - The test produced an error in running the test itself.
+     - Suite, Case
+   * - FAIL
+     - failed
+     - The test produced results different than expected.
+     - Suite, Case
+   * - PASS
+     - passed
+     - The test produced results as expected.
+     - Suite, Case
+   * - SKIP
+     - skipped
+     - Test was skipped by some other reason than previously delineated.
+     - Suite, Case
+
+**In-code** and **In-text** columns should the names of a given status.
+If one is not a Twister developer, the in-text name should suffice.
+It is the name used in the JSON reports.
+
+**Where applicable** column indicates where we might find such Statuses.
+Harness has been omitted for clarity, as it does not appear in final report files.
+If a status is marked as applicable for Suites, it is also applicable for Instances.
+
+.. note::
+
+   There are two more Statuses that are internal to Twister.
+   ``NONE`` (A starting status for Cases and Suites) and
+   ``STARTED`` (Indicating an in-progress Case).
+   Those should not appear in the final Twister report.
+   Appearance of non-terminal Statuses in one's report files indicates a problem with Twister.
+
+
+Test Case and Test Suite Status combinations
+============================================
+
+.. list-table:: Test Case and Test Suite Status combinations
+   :widths: 22 13 13 13 13 13 13
+   :align: center
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Case\\Suite
+     - FILTER
+     - ERROR
+     - FAIL
+     - PASS
+     - NOTRUN
+     - SKIP
+   * - FILTER
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+     - ✕
+     - ✕
+   * - ERROR
+     - ✕
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+     - ✕
+   * - BLOCK
+     - ✕
+     - ✓
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+   * - FAIL
+     - ✕
+     - ✓
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+   * - PASS
+     - ✕
+     - ✓
+     - ✓
+     - ✓
+     - ✕
+     - ✕
+   * - NOTRUN
+     - ✕
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✕
+   * - SKIP
+     - ✕
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+✕ indicates that such a combination should not happen in a proper Twister run. In other words,
+no Suite of a status indicated by the table column should contain any Cases of a status indicated
+by the table row. ✓ indicates a proper combination.
+
+Detailed explanation, per Test Suite Status
```

Comment:
Isn't this paragraph almost the same as what is in the 'Possible Twister Statuses' table and can be moved there ?
If a state meaning is different for TestSuite vs. TestCase they can be split into table columns.

Reply:
While the first sentence, the general Status description, is very similar to the table, in this section I aimed to explain why some combinations are possible and others are not.
When a documentation reader encounters the combination table, I'd assume they'd like to ask *why* certain combinations are impossible. So this needs to be after the combination table.

Reply:
unfortunately, having two descriptions tend to diverge from each other not clarifying much,
for example `FILTER`:
* `A static or runtime filter has eliminated the test from the list of tests to use.`
* `This status indicates that the whole Suite has been statically filtered out of a given Twister run. Thus, any Case within it should also have such a status.`
The second sentence don't mention runtime filtering and it is still vague why other combinations are impossible (if it is the goal to answer here), e.g. together with `PASS` why Suite: PASS and Case:FILTER is impossible ? same for `NOTRUN`.

`SKIP` is another example as it explains only one combination, isn't it ?